src/drivers/express/ExpressRouteAuthDriver.ts

```
@@ -0,0 +1,30 @@
+import { Router } from 'express';
+import * as interactor from '../../maintenance/maintenanceInteractor';
+
+export class ExpressRouteAuthDriver {
+    public static buildRouter(): Router {
+      const e = new ExpressRouteAuthDriver();
+      const router: Router = Router();
+      e.setRoutes(router);
+      return router;
+    }
+
+    private setRoutes(router: Router): void {
+    // SET CLARK MAINTENANCE NOTIFICATION
+    router.post('/maintenance', async(req, res) => {
+        try {
+          const val = req.body.clarkDown;
+          const user = req.user;
+          if (user.accessGroups !== undefined && user.accessGroups.includes('admin')) {
```

This logic should be exported to a helper function. It can be rewritten as `if/else`. No need for `else if` here.

---

If not authorized, the function should throw an error. The error will then be handled by the route handler `try/catch`

---

Updated